internal/agenthttp/client.go

```diff
@@ -26,6 +30,29 @@ func NewClient(opts ...ClientOption) *http.Client {
 		opt(&conf)
 	}
 
+	// http client profile is used to switch between different http client implementations
+	// - stdlib: uses the standard library http client
+	switch conf.HTTPClientProfile {
+	case "stdlib":
```

Comment:
I think the one thing missing here is support for `AllowHTTP2: false`.

Reply:
@DrJosh9000 I put a note in the PR about this, as it is quite complex and already a feature of the STDLIB maybe we do it the standard library way?

> Disabling H2 is not provided as this is avaliable via the GODEBUG environment variable supported by the Go Team. https://pkg.go.dev/net/http#hdr-HTTP_2

Reply:
/me looks again

Ah, yep, carry on.